community/kernel/src/main/java/org/neo4j/kernel/impl/core/NodeProxy.java

```diff
@@ -119,6 +119,11 @@ public void delete()
             throw new IllegalStateException( "Auto indexing encountered a failure while deleting the node: "
                                              + e.getMessage(), e );
         }
+        catch ( EntityNotFoundException e )
+        {
+            throw new NotFoundException( "Unable to delete node[" +
+                                         getId() + "] since it is already deleted." );
+        }
```

Comment:
This should be handled by the `nodeDelete` return value, so we don't need this catch.